src/main/java/com/sap/oss/phosphor/fosstars/data/owasp/experimental/VulnerabilitiesFromOwaspDependencyCheck.java

```diff
@@ -0,0 +1,404 @@
+  public ValueCache<MavenArtifact> cache() {
+    return NoValueCache.create();
+  }
+
+  /**
+   * There is not call back required for this data provider.
+   */
+  @Override
+  public DataProvider<MavenArtifact> set(UserCallback callback) {
+    return this;
+  }
+
+  /**
+   * No cache value is needed that is used by the data provider.
+   */
+  @Override
+  public DataProvider<MavenArtifact> set(ValueCache<MavenArtifact> cache) {
+    return this;
+  }
+
+  /**
+   * No configuration is required for this data provider.
+   */
+  @Override
+  public DataProvider<MavenArtifact> configure(Path config) throws IOException {
+    return this;
+  }
+
+  @Override
+  public DataProvider<MavenArtifact> update(MavenArtifact artifact, ValueSet values)
+      throws IOException {
+    Objects.requireNonNull(artifact, "On no! The artifact object cannot be null");
+    if (!artifact.version().isPresent()) {
+      throw new IOException("Oh no! the version is not available.");
+    }
+
+    Optional<OwaspDependencyCheckEntry> owaspDependencyCheckEntry = scan(artifact);
+    if (!owaspDependencyCheckEntry.isPresent()) {
+      values.update(VULNERABILITIES.unknown());
+      return this;
+    }
+
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    for (Dependency dependency : owaspDependencyCheckEntry.get().getDependencies()) {
+      if (dependency.getVulnerabilities() == null) {
+        continue;
+      }
+
+      for (OwaspDependencyCheckVuln owaspDependencyCheckvuln : dependency.getVulnerabilities()) {
+        vulnerabilities.add(from(owaspDependencyCheckvuln));
+      }
+    }
+
+    values.update(VULNERABILITIES.value(vulnerabilities));
+    return this;
+  }
+
+  /**
+   * Returns the supported feature loaded by this data provider.
+   */
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return setOf(VULNERABILITIES);
+  }
+
+  /**
+   * Scan the input jar file and analyze the extracted {@link Dependency}.
+   * 
+   * @param file The jar.
+   */
+  private void analyze(File file) {
+    try {
+      engine.scan(file);
+      engine.analyzeDependencies();
+    } catch (ExceptionCollection e) {
+      exceptionCollection.addException(e);
+    }
+  }
+
+  /**
+   * Process the report generated from analysis of the jar file.
+   * 
+   * @param fileName The name of report.
+   * @return An optional of {@link OwaspDependencyCheckEntry}.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<OwaspDependencyCheckEntry> process(String fileName) throws IOException {
+    Optional<Path> reportPath = createDirectory(REPORT_DIR);
+
+    if (reportPath.isPresent()) {
+      File file = reportPath.get().resolve(String.format("%s.json", fileName)).toFile();
+
+      try {
+        engine.writeReports(fileName, file, REPORT_OUTPUT_FORMAT, exceptionCollection);
+        ObjectMapper mapper = new ObjectMapper();
+        return Optional.ofNullable(mapper.readValue(file, OwaspDependencyCheckEntry.class));
```

Comment:
Please always use `Json.mapper()` to get an `ObjectMapper`. The `Json` class returns a safe `ObjectMapper`.

```suggestion
        return Optional.ofNullable(Json.mapper().readValue(file, OwaspDependencyCheckEntry.class));
```